support/cas-server-support-oauth/src/test/java/org/apereo/cas/support/oauth/validator/DefaultOAuth20ClientSecretValidatorTests.java

```diff
@@ -59,4 +60,27 @@ public void verifyClientSecretUndefined() {
         val result = oauth20ClientSecretValidator.validate(registeredService, secret);
         assertTrue(result);
     }
+
+    @Test
+    public void verifyClientSecretUrlEncoded() {
+        val secret = "!@#$%^&^&*()";
+        val encodedSecret = EncodingUtils.urlEncode(secret);
+        val registeredService = new OAuthRegisteredService();
+        registeredService.setClientId("clientid");
+        registeredService.setClientSecret(secret);
+        val result = oauth20ClientSecretValidator.validate(registeredService, encodedSecret);
+        assertTrue(result);
+    }
+
+    @Test
+    public void verifyNullClientSecretUrlEncoded() {
+        val secret = "!@#$%^&^&*()";
+        val registeredService = new OAuthRegisteredService();
+        registeredService.setClientId("clientid");
+        registeredService.setClientSecret(secret);
+        val result = oauth20ClientSecretValidator.validate(registeredService, null);
+        assertFalse(result);
+        val result2 = oauth20ClientSecretValidator.validate(registeredService, String.EMPTY);
```

Comment:
Are you sure `String.EMPTY` compiles?

Reply:
I must have compiled and run that in my dreams, fixed now.

Reply:
Thank you!